From: Roland Dreier <roland@purestorage.com>
Date: Wed, 28 May 2014 09:23:03 -0700
Subject: [PATCH 099/117] IB/mlx5: Fix warning about cast of wr_id back to
 pointer on 32 bits
Patch-mainline: v3.16-rc1
Git-commit: 6c9b5d9b00ed2b1cbd5e5d2c176bf88da7beb224
References: bsc#923036 FATE#318772

We need to cast wr_id to unsigned long before casting to a pointer.
This fixes:

       drivers/infiniband/hw/mlx5/mr.c: In function 'mlx5_umr_cq_handler':
    >> drivers/infiniband/hw/mlx5/mr.c:724:13: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
          context = (struct mlx5_ib_umr_context *)wc.wr_id;

Reported-by: kbuild test robot <fengguang.wu@intel.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
Signed-off-by: Hadar Hen Zion <hadarh@mellanox.com>
Acked-by: Cho, Yu-Chen <acho@suse.com>
---
 drivers/infiniband/hw/mlx5/mr.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/infiniband/hw/mlx5/mr.c
+++ b/drivers/infiniband/hw/mlx5/mr.c
@@ -721,7 +721,7 @@ void mlx5_umr_cq_handler(struct ib_cq *c
 		if (err == 0)
 			break;
 
-		context = (struct mlx5_ib_umr_context *)wc.wr_id;
+		context = (struct mlx5_ib_umr_context *) (unsigned long) wc.wr_id;
 		context->status = wc.status;
 		complete(&context->done);
 	}
